src/lib/swapper/swappers/ThorchainSwapper/endpoints.ts

```diff
@@ -31,37 +56,173 @@ export const thorchainApi: SwapperApi = {
     })
   },
 
-  getUnsignedTx: async ({
-    accountMetadata,
-    tradeQuote,
+  getUnsignedEvmTransaction: async ({
+    chainId,
     from,
+    nonce,
+    tradeQuote,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    // TODO: pull these from db using id so we don't have type zoo and casting hell
+    const { router: to, data, steps } = tradeQuote as ThorEvmTradeQuote
+    const { sellAmountIncludingProtocolFeesCryptoBaseUnit, sellAsset } = steps[0]
+
+    const value = isNativeEvmAsset(sellAsset.assetId)
+      ? sellAmountIncludingProtocolFeesCryptoBaseUnit
+      : '0'
+
+    const api = (() => {
+      switch (chainId) {
+        case KnownChainIds.EthereumMainnet:
+          return new evm.ethereum.V1Api(
+            new evm.ethereum.Configuration({
+              basePath: getConfig().REACT_APP_UNCHAINED_ETHEREUM_HTTP_URL,
+            }),
+          )
+        case KnownChainIds.AvalancheMainnet:
+          return new evm.avalanche.V1Api(
+            new evm.avalanche.Configuration({
+              basePath: getConfig().REACT_APP_UNCHAINED_AVALANCHE_HTTP_URL,
+            }),
+          )
+        default:
+          throw Error(`Unsupported chainId '${chainId}'`)
+      }
+    })()
```

Comment:
As discussed, we could import these, but the rationale is decoupling from chain-adapters dependencies, which this achieves ✅